single-kernel/median.cpp

```diff
@@ -64,8 +64,8 @@ class MedianFilterBench
           int k = 0;
           for(int i = -1; i<2; i++)
             for(int j = -1; j<2; j++) {
-              uint xs = fmin(fmax(x+j, 0), size-1); // borders are handled here with extended values
-              uint ys = fmin(fmax(y+i, 0), size-1);
+              uint xs = s::min(s::max(x+j, 0), static_cast<int>(size-1)); // borders are handled here with extended values
```

Comment:
This now produces a `error: reference to __device__ function 'min' in __host__ __device__ function` for me, using Clang 8 and hipSYCL@master.

Sorry for only getting back to you on this now, but I didn't realize this PR would include changes to kernels.

Reply:
I did this as a quick change since using fmin in kernels is not allowed. I can reproduce the issue, seems we have found a bug in the hipSYCL headers.

Reply:
Well, I'm glad it's reproducible, unlike issues we had in the past ;-).

Reply:
I think this is fixed by this PR: https://github.com/illuhad/hipSYCL/pull/199